packages/firebase_messaging/android/src/main/java/io/flutter/plugins/firebasemessaging/FlutterFirebaseMessagingService.java

```diff
@@ -18,16 +39,81 @@
   public static final String ACTION_TOKEN = "io.flutter.plugins.firebasemessaging.TOKEN";
   public static final String EXTRA_TOKEN = "token";
 
+  private static final String SHARED_PREFERENCES_KEY = "io.flutter.android_fcm_plugin";
+  private static final String BACKGROUND_SETUP_CALLBACK_HANDLE_KEY = "background_setup_callback";
+  private static final String BACKGROUND_MESSAGE_CALLBACK_HANDLE_KEY =
+      "background_message_callback";
+
+  // TODO(kroikie): make sIsIsolateRunning per-instance, not static.
+  private static AtomicBoolean sIsIsolateRunning = new AtomicBoolean(false);
+
+  /** Background Dart execution context. */
+  private static FlutterNativeView sBackgroundFlutterView;
+
+  private static MethodChannel sBackgroundChannel;
+
+  private static Long sBackgroundMessageHandle;
+
+  private static List<RemoteMessage> sBackgroundMessageQueue =
+      Collections.synchronizedList(new LinkedList<RemoteMessage>());
+
+  private static PluginRegistry.PluginRegistrantCallback sPluginRegistrantCallback;
+
+  private static final String TAG = "FlutterFcmService";
```

Comment:
Nit: I think this should generally match the class you're in, i.e. `FlutterFirebaseMessagingService`

Reply:
Done, I seem to remember there being a string length limit on Tags for logcat but maybe that restriction has been removed.

Reply:
Looks like 23 is still the max so reverting to `FlutterFcmService`.